src/vuestic-theme/vuestic-components/va-table/VaTable.vue

```diff
@@ -0,0 +1,251 @@
+  name: 'va-table',
+  components: {
+    SpringSpinner,
+    Vuetable,
+    VaPagination,
+  },
+  props: {
+    fields: {
+      type: Array,
+      required: true,
+    },
+    data: {
+      type: Array,
+      required: true,
+    },
+    perPage: {
+      type: Number,
+      default: 6,
+    },
+    apiMode: Boolean,
+    clickable: Boolean,
+    noPagination: Boolean,
+    noDataLabel: {
+      type: String,
+      default: undefined,
+    },
+    rowClass: {
+      type: Function,
+      default: undefined,
+    },
+    totalPages: {
+      type: Number,
+      default: 0,
+    },
+    loading: Boolean,
+  },
+  data () {
+    return {
+      currentPage: 1,
+    }
+  },
+  computed: {
+    styles () {
+      return {
+        tableClass: 'va-table' + (this.clickable ? ' clickable' : ''),
+        ascendingIcon: 'fa fa-caret-up',
+        descendingIcon: 'fa fa-caret-down',
+        renderIcon: classes => {
+          return '<span class="' + classes.join(' ') + '"></span>'
+        },
+      }
+    },
+    paginationTotal () {
+      return this.apiMode ? this.totalPages : Math.ceil(this.data.length / this.perPage)
+    },
+    paginationVisible () {
+      if (this.paginationTotal > 4) {
+        return 4
+      }
+
+      return this.paginationTotal
+    },
+  },
+  watch: {
+    perPage () {
+      this.refresh()
+    },
+    data () {
+      this.refresh()
+    },
+  },
+  methods: {
+    dataManager (sortOrder, pagination) {
+      let sorted = []
+
+      if (!sortOrder.length) {
+        sorted = this.data
+      } else {
+        const { sortField, direction } = sortOrder[0]
+        sorted = direction === 'asc' ? this.sortAsc(this.data, sortField) : this.sortDesc(this.data, sortField)
+      }
+
+      pagination = this.buildPagination(sorted.length, this.perPage)
+      const { from } = pagination
+      const sliceFrom = from - 1
+
+      return {
+        pagination,
+        data: sorted.slice(sliceFrom, sliceFrom + this.perPage),
+      }
+    },
+    sortAsc (items, field) {
+      return items.sort((a, b) => {
+        return a[field].localeCompare(b[field])
+      })
+    },
+    sortDesc (items, field) {
+      return items.sort((a, b) => {
+        return b[field].localeCompare(a[field])
+      })
+    },
+    buildPagination (l, perPage) {
+      return this.$refs.vuetable.makePagination(l, perPage)
+    },
+    inputPage (page) {
+      if (this.apiMode) {
+        this.$emit('page-selected', page)
+        return
+      }
+
+      this.$refs.vuetable.changePage(page)
+    },
+    refresh () {
+      this.$refs.vuetable.refresh()
+    },
+    rowClicked (row) {
+      this.$emit('row-clicked', row)
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+  @import "../../vuestic-sass/resources/resources";
+
+  .loading-container {
+    position: relative;
+  }
+
+  .loading-overlay {
+    position: absolute;
+    background: rgba(255,255,255,0.5);
+    top: 0;
+    bottom: 0;
+    width: 100%;
+  }
+
+  .va-table {
```

Comment:
Normally this class is used for higher level parent in DOM.
In our case it's unclassed div.

For table you can use
`.va-table__vuetable`

Reply:
done